contracts/modules/cover/Cover.sol

```diff
@@ -47,18 +43,65 @@ contract Cover is ICover, MasterAwareV2 {
   uint public constant STAKE_SPEED_UNIT = 100000e18;
   uint public constant PRICE_CURVE_EXPONENT = 7;
   uint public constant MAX_PRICE_PERCENTAGE = 1e20;
+  IQuotationData internal immutable quotationData;
 
+  /* ========== CONSTRUCTOR ========== */
 
-  constructor() {
+  constructor(IQuotationData _quotationData, IProductsV1 _productsV1) {
+    quotationData = _quotationData;
+    productsV1 = _productsV1;
   }
 
-  function initialize(ICoverNFT _coverNFT) public {
-    require(address(coverNFT) == address(0), "Cover: already initialized");
+  function initialize(address _coverNFT) public {
+    require(coverNFT == address(0), "Cover: already initialized");
     coverNFT = _coverNFT;
   }
 
   /* === MUTATIVE FUNCTIONS ==== */
 
+  // @dev Migrates covers from V1 to Cover.sol
+  function migrateCover(uint coverId) external {
+    (
+      /*uint coverId*/,
+      address coverOwner,
+      address legacyProductId,
+      bytes4 currencyCode,
+      /*uint sumAssured*/,
+      uint premiumNXM
+    ) = quotationData.getCoverDetailsByCoverID1(coverId);
+    (
+      /*uint coverId*/,
+      uint8 status,
+      uint sumAssured,
+      uint16 coverPeriodInDays,
+      uint validUntil
+    ) = quotationData.getCoverDetailsByCoverID2(coverId);
+
+    require(msg.sender == coverOwner, "Cover can only be migrated by its owner");
```

Comment:
so because of the check the end users need to call this through the distributor contract for covers boguht that way correct? arNFT covers

Reply:
Correct. The distributor is the current owner of all covers bought through them.